balancer/weightedroundrobin/balancer.go

```diff
@@ -595,32 +593,34 @@ func (w *weightedSubConn) updateConnectivityState(cs connectivity.State) connect
 // will cause the backend weight to be treated as the mean of the weights of the
 // other backends. If forScheduler is set to true, this function will emit
 // metrics through the mtrics registry.
-func (w *weightedSubConn) weight(now time.Time, weightExpirationPeriod, blackoutPeriod time.Duration, forScheduler bool) (weight float64) {
+func (w *weightedSubConn) weight(now time.Time, weightExpirationPeriod, blackoutPeriod time.Duration, recordMetrics bool) float64 {
 	w.mu.Lock()
 	defer w.mu.Unlock()
 
-	weight = 0
-	if forScheduler {
-		defer endpointWeightsHandle.Record(w.metricsRecorder, weight, []string{w.target, w.locality}...)
+	weight := float64(0)
+	if recordMetrics {
+		defer func() {
+			endpointWeightsMetric.Record(w.metricsRecorder, weight, w.target, w.locality)
+		}()
 	}
 
 	// If the most recent update was longer ago than the expiration period,
 	// reset nonEmptySince so that we apply the blackout period again if we
 	// start getting data again in the future, and return 0.
 	if now.Sub(w.lastUpdated) >= weightExpirationPeriod {
-		if forScheduler {
-			endpointWeightStaleHandle.Record(w.metricsRecorder, 1, []string{w.target, w.locality}...)
+		if recordMetrics {
+			endpointWeightStaleMetric.Record(w.metricsRecorder, 1, w.target, w.locality)
 		}
 		w.nonEmptySince = time.Time{}
-		return
+		return weight
 	}
 
 	// If we don't have at least blackoutPeriod worth of data, return 0.
 	if blackoutPeriod != 0 && (w.nonEmptySince == (time.Time{}) || now.Sub(w.nonEmptySince) < blackoutPeriod) {
-		endpointWeightNotYetUsableHandle.Record(w.metricsRecorder, 1, []string{w.target, w.locality}...)
+		endpointWeightNotYetUsableMetric.Record(w.metricsRecorder, 1, w.target, w.locality)
 
-		return
+		return weight
 	}
-
-	return w.weightVal
+	weight = w.weightVal
```

Comment:
Yeah...I'd prefer to use what you had before.  Use the named return value, and don't ever assign to `weight` at all.  Just read it in the defer to find out what was returned.

Reply:
Switched to named return value.